mod/friends/classes/Elgg/Friends/Menus/Filter.php

```diff
@@ -1,24 +1,51 @@
 <?php
 
-namespace Elgg\Friends;
-
-use Elgg\Menu\MenuItems;
+namespace Elgg\Friends\Menus;
 
 /**
- * Add tabs to the filter menu in the friends context
+ * Hook callbacks for menus
+ *
+ * @since 4.0
  *
- * @since 3.2
+ * @internal
  */
-class FilterMenu {
+class Filter {
 
+	/**
+	 * Add 'Friends' tab to common filter
+	 *
+	 * @param \Elgg\Hook $hook 'filter_tabs', 'all'
+	 *
+	 * @return void|ElggMenuItem[]
+	 */
+	public static function registerFilterTabs(\Elgg\Hook $hook) {
+	
+		$user = $hook->getUserParam();
+		if (!$user instanceof \ElggUser) {
+			return;
+		}
+	
+		$vars = $hook->getParam('vars');
+		$selected = $hook->getParam('selected');
+		$type = $hook->getType();
+	
+		$items = $hook->getValue();
+		$items[] = \ElggMenuItem::factory([
+			'name' => 'friend',
+			'text' => elgg_echo('friends'),
+			'href' => (isset($vars['friend_link'])) ? $vars['friend_link'] : "$type/friends/{$user->username}",
```

Comment:
shouldn't we add a check for friends route? No idea if enough information is present

Reply:
no idea

Reply:
#13093